### ISSUE TITLE
Az.resources.testsupport issue

### ISSUE THREAD

**Issue by dicolanl** [Azure/azure-powershell#14657]
## Description
following the wiki to auto generate powershell code for an RP.  I am doing this in the docker image for autorest.

running test-module.ps1 errors with
```powershell
root@e1219b621a59:/src/SecurityInsights# pwsh test-module.ps1 -live
Creating isolated process...
Deploying resourceGroup:  azp-sentinel-0bfc8h

Deploying workspace:  azp-sentinel-ws-0jonib
New-AzDeployment : A parameter cannot be found that matches parameter name 'ResourceGroupName'.
At /src/SecurityInsights/test/utils.ps1:32 char:5
+     New-AzDeployment -Name workspace -ResourceGroupName $RGName -Mode ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidArgument: (:) [New-AzDeployment], ParameterBindingException
+ FullyQualifiedErrorId : NamedParameterNotFound,New-AzDeployment
```

i did get-help new-azdeployment command to confirm parameters.
```powershell

NAME
    New-AzDeployment
    
SYNOPSIS
    You can provide the template and parameters directly in the request or link to JSON files.
    
    
SYNTAX
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> -TemplateFile <String> -TemplateParameterFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> -TemplateFile <String> -TemplateParameterFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> -TemplateFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> -TemplateFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterObject <Hashtable> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> -TemplateParameterFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterJson <String> -TemplateJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterObject <Hashtable> -TemplateJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> -TemplateParameterFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateObject <Hashtable> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateObject <Hashtable> -TemplateParameterJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -ResourceGroupName <String> -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateObject <Hashtable> -TemplateParameterObject <Hashtable> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> -TemplateFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> -TemplateFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterObject <Hashtable> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> -TemplateParameterFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateObject <Hashtable> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> -TemplateParameterFile <String> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateObject <Hashtable> -TemplateParameterJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateObject <Hashtable> -TemplateParameterObject <Hashtable> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterObject <Hashtable> -TemplateJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    New-AzDeployment [-SubscriptionId <String>] [-Name <String>] -Mode <DeploymentMode> [-DeploymentDebugLogLevel <String>] [-Location <String>] -TemplateParameterJson <String> -TemplateJson <String> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-WhatIf] [-Confirm] [<CommonParameters>]
    
    
DESCRIPTION
    You can provide the template and parameters directly in the request or link to JSON files.
    

RELATED LINKS
    https://docs.microsoft.com/en-us/powershell/module/az.resources.testsupport/new-azdeployment

REMARKS
    To see the examples, type: "get-help New-AzDeployment -examples".
    For more information, type: "get-help New-AzDeployment -detailed".
    For technical information, type: "get-help New-AzDeployment -full".
    For online help, type: "get-help New-AzDeployment -online"
```

my snippet from utils.ps1
```powershell
Write-Host "Deploying workspace: "$WorkspaceName
    $workspaceParams = Get-Content .\test\deploymentTemplates\workspace\azuredeploy.parameters.json | ConvertFrom-Json
    $workspaceParams.parameters.workspaceName.value = $WorkspaceName
    Set-Content -Path .\test\deploymentTemplates\workspace\azuredeploy.parameters.json -Value (ConvertTo-Json $workspaceParams)
    $templateFile = Get-ChildItem .\test\deploymentTemplates\workspace\azuredeploy.json
    $templateParameterFile = Get-ChildItem .\test\deploymentTemplates\workspace\azuredeploy.parameters.json
    New-AzDeployment -Name workspace -ResourceGroupName $RGName -Mode Incremental -TemplateFile ($templateFile.FullName) -TemplateParameterFile ($templateParameterFile.FullName)
```

**Comment by dicolanl** [comment]
i thought i might be a docker issue.  but i tried this in windows too

```powershell
PS E:\azure-powershell-generated\src\SecurityInsights> New-AzDeployment -Name workspace -ResourceGroupName azp-sentinel-bcu7tl -Mode Incremental -TemplateFile E:\azure-powershell-generated\src\SecurityInsights\test\deploymentTemplates\workspace\azuredeploy.json -TemplateParameterFile E:\azure-powershell-generated\src\SecurityInsights\test\deploymentTemplates\workspace\azuredeploy.parameters.json
New-AzDeployment: A parameter cannot be found that matches parameter name 'ResourceGroupName'.
```

**Comment by dingmeng-xue** [comment]
@dicolanl , `New-AzDeployment` is deployment on subscription level. So, it doesn't support resource group name. If you want to go to ResourceGroup level deployment, `New-AzResourceGroupDeployment` should be a correct one.

**Comment by dicolanl** [comment]
@dingmeng-xue  per the (wiki)[https://github.com/Azure/azure-powershell/wiki/How-to-On-Board-New-RP-with-Azure-PowerShell-Generator#test]  a special resources module is used during testing. that module does not have new-azresourcegroupdeployment

```powershell
PS E:\azure-powershell-generation\src\SecurityInsights [Az.SecurityInsights]> import-module C:\Users\ndicola\.PSSharedModules\Resources\Az.Resources.TestSupport.psd1
PS E:\azure-powershell-generation\src\SecurityInsights [Az.SecurityInsights]> get-command -Module Az.Resources.TestSupport

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Alias           Get-AzResourceGroupDeployment                      0.0.1      Az.Resources.TestSupport
Alias           Test-AzResourceGroupExistence                      0.0.1      Az.Resources.TestSupport
Function        Get-AzDeployment                                   0.0.1      Az.Resources.TestSupport
Function        Get-AzResourceGroup                                0.0.1      Az.Resources.TestSupport
Function        Get-AzResourceProvider                             0.0.1      Az.Resources.TestSupport
Function        New-AzDeployment                                   0.0.1      Az.Resources.TestSupport
Function        New-AzResourceGroup                                0.0.1      Az.Resources.TestSupport
Function        Remove-AzResourceGroup                             0.0.1      Az.Resources.TestSupport
Function        Test-AzResourceGroup                               0.0.1      Az.Resources.TestSupport
```

**Comment by dolauli** [comment]
Fixed in autorest/powershell@3.0.426. To use this version, you may need to run `'autoreset --reset'` to clean the old versions.
